### ISSUE TITLE
Generator validation loss is significantly different from the sum of all the validation losses

### ISSUE THREAD

**Issue by maxpv** [keras-team/keras#14044]
**System information**  
- TensorFlow backend (yes / no):  yes
- TensorFlow version: 2.2.0 
- Keras version:  2.3.1
- Python version:  3.6.9

Working with a 12 heads output model I noticed that my model checkpoints were not saved by the callback. I then realized that the validation loss wasn't decreasing despite the fact that all the individual validation losses are, I measure a validation loss of 0.345047 while the sum of all the individual losses at 0.662322.

While I expect to have a slight difference due to floating-point operations I wasn't expecting such one. I tried to reproduce the issue and noticed that this happens while using _generators and on the validation loss only_. 

You can reproduce similar results on this [notebook](https://colab.research.google.com/drive/1mfcscNUFzAA5_5mxwuZuboPciiEYt0Q9?usp=sharing), notice the 10^3 difference between the `val_diff` and the `train_diff`:

trained by| val_loss | sum_val_loss | loss | sum_train_loss | val_diff | train_diff
-- | -- | -- | -- | -- | -- | --
array | 8.347330 | 8.347331 | 8.444574 | 8.444574 | -1.549721e-07 | 3.886223e-07
generator | 8.348998 | 8.349756 | 8.442474 | 8.442474 | -0.000758 | -7.152557e-08

`val_loss` is the validation loss computed by keras
`sum_val_loss` is the sum of all the validation losses for each model's heads
`val_diff` is the difference between the two

And similarly for the training loss.

**Comment by maxpv** [comment]
This is not a support question, replace `keras` by `tensorflow.keras` and the problem disappear.

**Comment by mikus** [comment]
I have observed similar situation. Moreover, reported validation loss somehow fluctuates while summary validation loss is quite stable. I also used `fit_generator`.

<img width="895" alt="image" src="https://user-images.githubusercontent.com/834701/83954982-5da17800-a84e-11ea-8330-d4816bcd1b30.png">

##### System information

    TensorFlow backend (yes / no): yes
    TensorFlow version: 1.15.3
    Keras version: 2.3.1
    Python version: 3.6.6
